service/controller/resource_set.go

```diff
@@ -403,7 +403,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.VersionBundleVersion(customObject) == project.BundleVersion() {
+		if key.OperatorVersion(customObject) == project.BundleVersion() {
```

Comment:
This is the most importante change. Now this operator will only handle `CustomResources` that contain a label with the version that matches the operator version. It won't use the `spec` field.